src/Microsoft.Framework.Runtime.Roslyn/AfterCompileContext.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.Framework.Runtime.Roslyn
+{
+    internal class AfterCompileContext : IAfterCompileContext
+    {
+        private readonly CompilationContext _context;
+
+        public AfterCompileContext(CompilationContext context, IEnumerable<Diagnostic> emitDiagnostics)
+        {
+            _context = context;
+            var diagnostics = new List<Diagnostic>(context.Diagnostics);
+            diagnostics.AddRange(emitDiagnostics);
+
+            Diagnostics = diagnostics;
+        }
+
+        public IProjectContext ProjectContext
+        {
+            get
+            {
+                return _context.ProjectContext;
+            }
+        }
+
+        public Stream AssemblyStream { get; set; }
```

Comment:
It avoids us having to do the first emit. Just a thought since its quite likely you're going to have to emit it if the resulting Assembly is interesting. Quite unlikely you're going to look at the Stream does much.